test/unit/app/filteringTest.js

```diff
@@ -0,0 +1,101 @@
+/* global describe, before, after, it */
+const mockery = require('mockery')
+const assert = require('assert')
+const sinon = require('sinon')
+const {cookieExceptions, refererExceptions} = require('../../../js/data/siteHacks')
+
+require('../braveUnit')
+
+describe('filtering unit tests', function () {
+  let filtering
+  const fakeElectron = require('../lib/fakeElectron')
+
+  before(function () {
+    mockery.enable({
+      warnOnReplace: false,
+      warnOnUnregistered: false,
+      useCleanCache: true
+    })
+    mockery.registerMock('electron', fakeElectron)
+    mockery.registerMock('./adBlock', {adBlockResourceName: 'adblock'})
+    filtering = require('../../../app/filtering')
+  })
+
+  after(function () {
+    mockery.disable()
+  })
+
+  describe('considerRequestExceptions', function () {
+    describe('when cookieSetting === "blockAllCookies"', function () {
+      let isResourceEnabledStub
+      before(function () {
+        isResourceEnabledStub = sinon.stub(filtering, 'isResourceEnabled').returns('blockAllCookies')
+      })
+      after(function () {
+        isResourceEnabledStub.restore()
+      })
+
+      it('clears cookie field', function () {
+        const url = 'https://cdnp3.stackassets.com/574db2390a12942fcef927356dadc6f9955edea9/store/fe3eb8fc014a20f2d25810b3c4f4b5b0db8695adfd7e8953721a55c51b90/sale_7217_primary_image.jpg'
+        const firstPartyUrl = 'https://slashdot.org/'
+        const requestHeaders = {
+          Cookie: 'optimizelyEndUserId=oeu1491721215718r0.024789086462633003; __ssid=97b17d31-8f1b-4193-8914-df36e7b740f6; optimizelySegments=%7B%22300150879%22%3A%22false%22%2C%22300333436%22%3A%22gc%22%2C%22300387578%22%3A%22campaign%22%7D; optimizelyBuckets=%7B%7D; _pk_id.40.2105=8fca10ea565f58bf.1485982886.187.1499406000.1499405260.; _pk_ses.40.2105=*'
+        }
+        const result = filtering.considerRequestExceptions(requestHeaders, url, firstPartyUrl, false)
+
+        assert.equal(result.Cookie, undefined)
+      })
+
+      describe('when there is a cookie exception', function () {
+        it('keeps the cookie field', function () {
+          const cookieException = cookieExceptions && cookieExceptions[0]
+
+          assert(cookieException)
+
+          const url = cookieException[0]
+          const firstPartyUrl = cookieException[1]
```

Comment:
The firstPartyUrl should be cookieException[0]. I will make it right in the commit changing the cookie exception to map